clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/DataRecoveryClient.java

```diff
@@ -45,8 +52,35 @@ public DataRecoveryMonitor getMonitor() {
     return monitor;
   }
 
+  public void filterStoresWithOngoingRepush(Set<String> storesList, StoreRepushCommand.Params params) {
+    String url = params.getUrl();
+    ControllerClient cli = params.getPCtrlCliWithoutCluster();
+    LocalDateTime timestamp = params.getTimestamp();
+    String sourceFabric = params.getSourceFabric();
+    for (String s: storesList) {
+      String clusterName = cli.discoverCluster(s).getCluster();
+      try (ControllerClient parentCtrlCli = new ControllerClient(clusterName, url, Optional.empty())) {
+        StoreHealthAuditResponse storeHealthInfo = parentCtrlCli.listStorePushInfo(s, false);
+        RegionPushDetails regionPushDetails = storeHealthInfo.getRegionPushDetails().get(sourceFabric);
+        String latestTimestamp = regionPushDetails.getPushStartTimestamp();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSSSS");
+        LocalDateTime latestPushStartTime = LocalDateTime.parse(latestTimestamp, formatter);
+        if (latestPushStartTime.compareTo(timestamp) > 0) {
+          LOGGER.warn(
+              "Removing store " + s
+                  + " from stores list, as the has a push job started more recently than the input timestamp.");
+          storesList.remove(s);
+        }
```

Comment:
Two thoughts:
1.  it doesn't sound like a warning message.
2. we should looking for a better way to report, e.g. a summary of which stores are skipped and for what reason?

What do you think?

Reply:
I added a new system to report the skipped stores before the shutdown step in a block, with error message. Let me know how this looks :)